Trouble Installing node-rdkafka on Node.js v23

Installation Issue with node-rdkafka on Node.js v23

**Environment Information**
- OS: Ubuntu 20.04.6 LTS
- Node Version: v23
- NPM Version: 10.9.0
- node-rdkafka version: 3.1.1

**Steps to Reproduce**
1. Open a terminal.
2. Run the following command to install node-rdkafka:
   ```bash
   npm install node-rdkafka
   ```

**Additional Context**
- I have all the necessary build tools installed.
- The issue persists regardless of whether I try to install it globally or locally.
- I am able to install node-rdkafka in Node.js versions less than 23.

Bumping this. We are also seeing this issue. Here are the npm error logs.

```
0 verbose cli /Users/norald.alejo/.nvm/versions/node/v23.1.0/bin/node /Users/norald.alejo/.nvm/versions/node/v23.1.0/bin/npm
1 info using npm@10.9.0
2 info using node@v23.1.0
3 silly config load:file:/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/npmrc
4 silly config load:file:/Users/norald.alejo/Programming/aioc/test/.npmrc
5 silly config load:file:/Users/norald.alejo/.npmrc
6 silly config load:file:/Users/norald.alejo/.nvm/versions/node/v23.1.0/etc/npmrc
7 verbose title npm install node-rdkafka
8 verbose argv "install" "node-rdkafka"
9 verbose logfile logs-max:10 dir:/Users/norald.alejo/.npm/_logs/2024-11-06T21_22_22_655Z-
10 verbose logfile /Users/norald.alejo/.npm/_logs/2024-11-06T21_22_22_655Z-debug-0.log
11 silly logfile start cleaning logs, removing 1 files
12 silly packumentCache heap:8640266240 maxSize:2160066560 maxEntrySize:1080033280
13 silly logfile done cleaning log files
14 silly idealTree buildDeps
15 silly fetch manifest node-rdkafka@*
16 silly packumentCache full:https://registry.npmjs.org/node-rdkafka cache-miss
17 http fetch GET 200 https://registry.npmjs.org/node-rdkafka 209ms (cache miss)
18 silly packumentCache full:https://registry.npmjs.org/node-rdkafka set size:undefined disposed:false
19 silly placeDep ROOT node-rdkafka@3.1.1 OK for: test@1.0.0 want: *
20 silly fetch manifest bindings@^1.3.1
21 silly packumentCache full:https://registry.npmjs.org/bindings cache-miss
22 silly fetch manifest nan@^2.19.0
23 silly packumentCache full:https://registry.npmjs.org/nan cache-miss
24 http fetch GET 200 https://registry.npmjs.org/bindings 34ms (cache miss)
25 silly packumentCache full:https://registry.npmjs.org/bindings set size:undefined disposed:false
26 http fetch GET 200 https://registry.npmjs.org/nan 71ms (cache miss)
27 silly packumentCache full:https://registry.npmjs.org/nan set size:undefined disposed:false
28 silly placeDep ROOT bindings@1.5.0 OK for: node-rdkafka@3.1.1 want: ^1.3.1
29 silly placeDep ROOT nan@2.22.0 OK for: node-rdkafka@3.1.1 want: ^2.19.0
30 silly fetch manifest file-uri-to-path@1.0.0
31 silly packumentCache full:https://registry.npmjs.org/file-uri-to-path cache-miss
32 http fetch GET 200 https://registry.npmjs.org/file-uri-to-path 34ms (cache miss)
33 silly packumentCache full:https://registry.npmjs.org/file-uri-to-path set size:undefined disposed:false
34 silly placeDep ROOT file-uri-to-path@1.0.0 OK for: bindings@1.5.0 want: 1.0.0
35 silly reify moves {}
36 silly audit bulk request {
36 silly audit   'node-rdkafka': [ '3.1.1' ],
36 silly audit   bindings: [ '1.5.0' ],
36 silly audit   nan: [ '2.22.0' ],
36 silly audit   'file-uri-to-path': [ '1.0.0' ]
36 silly audit }
37 silly tarball no local data for file-uri-to-path@https://registry.npmjs.org/file-uri-to-path/-/file-uri-to-path-1.0.0.tgz. Extracting by manifest.
38 silly tarball no local data for bindings@https://registry.npmjs.org/bindings/-/bindings-1.5.0.tgz. Extracting by manifest.
39 silly tarball no local data for nan@https://registry.npmjs.org/nan/-/nan-2.22.0.tgz. Extracting by manifest.
40 silly tarball no local data for node-rdkafka@https://registry.npmjs.org/node-rdkafka/-/node-rdkafka-3.1.1.tgz. Extracting by manifest.
41 http fetch GET 200 https://registry.npmjs.org/bindings/-/bindings-1.5.0.tgz 45ms (cache miss)
42 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 93ms
43 silly audit report {}
44 http fetch GET 200 https://registry.npmjs.org/file-uri-to-path/-/file-uri-to-path-1.0.0.tgz 94ms (cache miss)
45 http fetch GET 200 https://registry.npmjs.org/nan/-/nan-2.22.0.tgz 174ms (cache miss)
46 http fetch GET 200 https://registry.npmjs.org/node-rdkafka/-/node-rdkafka-3.1.1.tgz 582ms (cache miss)
47 info run node-rdkafka@3.1.1 install node_modules/node-rdkafka node-gyp rebuild
48 info run node-rdkafka@3.1.1 install { code: 1, signal: null }
49 verbose stack Error: command failed
49 verbose stack     at promiseSpawn (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
49 verbose stack     at spawnWithShell (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
49 verbose stack     at promiseSpawn (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
49 verbose stack     at runScriptPkg (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:77:13)
49 verbose stack     at runScript (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
49 verbose stack     at /Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js:329:17
49 verbose stack     at run (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:67:22)
49 verbose stack     at /Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:84:9
49 verbose stack     at new Promise (<anonymous>)
49 verbose stack     at callLimit (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:35:69)
50 verbose pkgid node-rdkafka@3.1.1
51 error code 1
52 error path /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka
53 error command failed
54 error command sh -c node-gyp rebuild
55 error ACTION deps_librdkafka_gyp_librdkafka_target_configure deps/librdkafka/config.h
55 error checking for OS or distribution... [32mok^[(B[m (osx)
55 error checking for C compiler from CC env... [33mfailed^[(B[m
55 error checking for gcc (by command)... [32mok^[(B[m
55 error checking for C++ compiler from CXX env... [33mfailed^[(B[m
55 error checking for C++ compiler (g++)... [32mok^[(B[m
55 error checking executable ld... [32mok^[(B[m
55 error checking executable nm... [32mok^[(B[m
55 error checking executable objdump... [32mok^[(B[m
55 error checking executable strip... [32mok^[(B[m
55 error checking executable libtool... [32mok^[(B[m
55 error checking executable ranlib... [32mok^[(B[m
55 error checking for debug symbols compiler flag (-g...)... [32mok^[(B[m
55 error checking for pkgconfig (by command)... [32mok^[(B[m
55 error checking for install (by command)... [33mfailed^[(B[m
55 error checking for GNU ar... [33mfailed^[(B[m (disable)
55 error checking for PIC (by compile)... [32mok^[(B[m
55 error checking for GNU-compatible linker options... [33mfailed^[(B[m
55 error checking for OSX linker options... [32mok^[(B[m
55 error checking for GNU linker-script ld flag... [33mfailed^[(B[m
55 error checking for Solaris linker-script ld flag... [33mfailed^[(B[m (ignore)
55 error checking for __atomic_32 (by compile)... [32mok^[(B[m
55 error checking for __atomic_64 (by compile)... [32mok^[(B[m
55 error checking for socket (by compile)... [32mok^[(B[m
55 error parsing version '0x020503ff'... [32mok^[(B[m (2.5.3)
55 error checking for librt (by pkg-config)... [33mfailed^[(B[m
55 error checking for librt (by compile)... [33mfailed^[(B[m
55 error checking for libpthread (by pkg-config)... [33mfailed^[(B[m
55 error checking for libpthread (by compile)... [32mok^[(B[m
55 error checking for c11threads (by pkg-config)... [33mfailed^[(B[m
55 error checking for c11threads (by compile)... [33mfailed^[(B[m (disable)
55 error checking for libdl (by pkg-config)... [33mfailed^[(B[m
55 error checking for libdl (by compile)... [32mok^[(B[m
55 error checking for zlib (by pkg-config)... [32mok^[(B[m
55 error checking for libcrypto (by pkg-config)... [32mok^[(B[m
55 error checking for libssl (by pkg-config)... [32mok^[(B[m
55 error checking for libsasl2 (by pkg-config)... [33mfailed^[(B[m
55 error checking for libsasl2 (by compile)... [32mok^[(B[m
55 error checking for libzstd (by pkg-config)... [32mok^[(B[m
55 error checking for libcurl (by pkg-config)... [32mok^[(B[m
55 error checking for libm (by pkg-config)... [33mfailed^[(B[m
55 error checking for libm (by compile)... [32mok^[(B[m
55 error checking for liblz4 (by pkg-config)... [32mok^[(B[m
55 error checking for syslog (by compile)... [32mok^[(B[m
55 error checking for rapidjson (by compile)... [33mfailed^[(B[m (disable)
55 error checking for crc32chw (by compile)... [32mok^[(B[m
55 error checking for regex (by compile)... [32mok^[(B[m
55 error checking for rand_r (by compile)... [32mok^[(B[m
55 error checking for strndup (by compile)... [32mok^[(B[m
55 error checking for strlcpy (by compile)... [32mok^[(B[m
55 error checking for strerror_r (by compile)... [32mok^[(B[m
55 error checking for strcasestr (by compile)... [32mok^[(B[m
55 error checking for pthread_setname_gnu (by compile)... [33mfailed^[(B[m (disable)
55 error checking for pthread_setname_darwin (by compile)... [32mok^[(B[m
55 error checking for nm (by env NM)... [32mok^[(B[m (cached)
55 error checking for getrusage (by compile)... [32mok^[(B[m
55 error Generated Makefile.config
55 error Generated config.h
55 error
55 error Configuration summary:
55 error   prefix                   /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps
55 error   MKL_DISTRO               osx
55 error   SOLIB_EXT                .dylib
55 error   ARCH                     x86_64
55 error   CPU                      generic
55 error   GEN_PKG_CONFIG           y
55 error   libdir                   /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps
55 error   MKL_APP_NAME             librdkafka
55 error   MKL_APP_DESC_ONELINE     The Apache Kafka C/C++ library
55 error   LDFLAGS                  -L/Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps
55 error   CC                       gcc
55 error   CXX                      g++
55 error   LD                       ld
55 error   NM                       nm
55 error   OBJDUMP                  objdump
55 error   STRIP                    strip
55 error   LIBTOOL                  libtool
55 error   RANLIB                   ranlib
55 error   CPPFLAGS                 -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align
55 error   PKG_CONFIG               pkg-config
55 error   INSTALL                  /usr/bin/install
55 error   HAS_GNU_AR               n
55 error   LIB_LDFLAGS              -shared -dynamiclib -Wl,-install_name,$(DESTDIR)$(libdir)/$(LIBFILENAME)
55 error   RDKAFKA_VERSION_STR      2.5.3
55 error   MKL_APP_VERSION          2.5.3
55 error   LIBS                     -L/usr/local/Cellar/lz4/1.9.4/lib -llz4 -lm -lcurl -L/usr/local/opt/zstd/lib -lzstd -lsasl2 -L/usr/local/Cellar/openssl@3/3.3.1/lib -lssl -L/usr/local/Cellar/openssl@3/3.3.1/lib -lcrypto -lz -ldl -lpthread
55 error   MKL_PKGCONFIG_LIBS_PRIVATE -lm -lsasl2 -ldl -lpthread
55 error   MKL_PKGCONFIG_REQUIRES_PRIVATE zlib libcrypto libssl libzstd libcurl liblz4
55 error   CFLAGS                   -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include
55 error   MKL_PKGCONFIG_REQUIRES   zlib libcrypto libssl libzstd libcurl liblz4
55 error   CXXFLAGS                 -Wno-non-virtual-dtor
55 error   SYMDUMPER                $(NM) -g
55 error   MKL_DYNAMIC_LIBS         -llz4 -lm -lcurl -lzstd -lsasl2 -lssl -lcrypto -lz -ldl -lpthread
55 error   exec_prefix              /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps
55 error   bindir                   /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/bin
55 error   sbindir                  /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/sbin
55 error   libexecdir               /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/libexec
55 error   datadir                  /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/share
55 error   sysconfdir               /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/etc
55 error   sharedstatedir           /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/com
55 error   localstatedir            /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/var
55 error   runstatedir              /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/var/run
55 error   includedir               /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/include
55 error   infodir                  /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/info
55 error   mandir                   /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/man
55 error   BUILT_WITH               GCC GXX PKGCONFIG OSXLD LIBDL PLUGINS ZLIB SSL SASL_CYRUS ZSTD CURL HDRHISTOGRAM LZ4_EXT SYSLOG SNAPPY SOCKEM SASL_SCRAM SASL_OAUTHBEARER OAUTHBEARER_OIDC CRC32C_HW
55 error Generated config.cache
55 error
55 error Now type 'make' to build
55 error   TOUCH 11a9e3388a67e1ca5c31c1d8da49cb6d2714eb41.intermediate
55 error   ACTION deps_librdkafka_gyp_librdkafka_target_build_dependencies 11a9e3388a67e1ca5c31c1d8da49cb6d2714eb41.intermediate
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka.c -o rdkafka.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_broker.c -o rdkafka_broker.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_msg.c -o rdkafka_msg.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_topic.c -o rdkafka_topic.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_conf.c -o rdkafka_conf.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_timer.c -o rdkafka_timer.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_offset.c -o rdkafka_offset.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_transport.c -o rdkafka_transport.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_buf.c -o rdkafka_buf.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_queue.c -o rdkafka_queue.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_op.c -o rdkafka_op.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_request.c -o rdkafka_request.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_cgrp.c -o rdkafka_cgrp.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_pattern.c -o rdkafka_pattern.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_partition.c -o rdkafka_partition.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_subscription.c -o rdkafka_subscription.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_assignment.c -o rdkafka_assignment.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_assignor.c -o rdkafka_assignor.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_range_assignor.c -o rdkafka_range_assignor.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_roundrobin_assignor.c -o rdkafka_roundrobin_assignor.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sticky_assignor.c -o rdkafka_sticky_assignor.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_feature.c -o rdkafka_feature.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdcrc32.c -o rdcrc32.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c crc32c.c -o crc32c.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdmurmur2.c -o rdmurmur2.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdfnv1a.c -o rdfnv1a.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c cJSON.c -o cJSON.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdaddr.c -o rdaddr.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdrand.c -o rdrand.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdlist.c -o rdlist.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c tinycthread.c -o tinycthread.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c tinycthread_extra.c -o tinycthread_extra.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdlog.c -o rdlog.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdstring.c -o rdstring.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_event.c -o rdkafka_event.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_metadata.c -o rdkafka_metadata.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdregex.c -o rdregex.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdports.c -o rdports.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_metadata_cache.c -o rdkafka_metadata_cache.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdavl.c -o rdavl.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl.c -o rdkafka_sasl.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl_plain.c -o rdkafka_sasl_plain.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_interceptor.c -o rdkafka_interceptor.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_msgset_writer.c -o rdkafka_msgset_writer.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_msgset_reader.c -o rdkafka_msgset_reader.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_header.c -o rdkafka_header.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_admin.c -o rdkafka_admin.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_aux.c -o rdkafka_aux.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_background.c -o rdkafka_background.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_idempotence.c -o rdkafka_idempotence.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_cert.c -o rdkafka_cert.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_txnmgr.c -o rdkafka_txnmgr.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_coord.c -o rdkafka_coord.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdbase64.c -o rdbase64.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdvarint.c -o rdvarint.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdbuf.c -o rdbuf.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdmap.c -o rdmap.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdunittest.c -o rdunittest.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_mock.c -o rdkafka_mock.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_mock_handlers.c -o rdkafka_mock_handlers.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_mock_cgrp.c -o rdkafka_mock_cgrp.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_error.c -o rdkafka_error.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_fetcher.c -o rdkafka_fetcher.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_telemetry.c -o rdkafka_telemetry.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_telemetry_encode.c -o rdkafka_telemetry_encode.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_telemetry_decode.c -o rdkafka_telemetry_decode.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c nanopb/pb_encode.c -o nanopb/pb_encode.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c nanopb/pb_decode.c -o nanopb/pb_decode.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c nanopb/pb_common.c -o nanopb/pb_common.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c opentelemetry/metrics.pb.c -o opentelemetry/metrics.pb.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c opentelemetry/common.pb.c -o opentelemetry/common.pb.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c opentelemetry/resource.pb.c -o opentelemetry/resource.pb.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl_cyrus.c -o rdkafka_sasl_cyrus.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl_scram.c -o rdkafka_sasl_scram.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl_oauthbearer.c -o rdkafka_sasl_oauthbearer.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c snappy.c -o snappy.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdgz.c -o rdgz.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_zstd.c -o rdkafka_zstd.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdhdrhistogram.c -o rdhdrhistogram.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_ssl.c -o rdkafka_ssl.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdhttp.c -o rdhttp.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_sasl_oauthbearer_oidc.c -o rdkafka_sasl_oauthbearer_oidc.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_lz4.c -o rdkafka_lz4.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -O3 -c rdxxhash.c -o rdxxhash.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rddl.c -o rddl.o
55 error gcc -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -I. -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/Cellar/openssl@3/3.3.1/include -I/usr/local/opt/zstd/include  -I/usr/local/Cellar/lz4/1.9.4/include -c rdkafka_plugin.c -o rdkafka_plugin.o
55 error [33mCreating shared library librdkafka.1.dylib^[(B[m
55 error gcc -L/Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps -shared -dynamiclib -Wl,-install_name,/Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/deps/librdkafka.1.dylib rdkafka.o rdkafka_broker.o rdkafka_msg.o rdkafka_topic.o rdkafka_conf.o rdkafka_timer.o rdkafka_offset.o rdkafka_transport.o rdkafka_buf.o rdkafka_queue.o rdkafka_op.o rdkafka_request.o rdkafka_cgrp.o rdkafka_pattern.o rdkafka_partition.o rdkafka_subscription.o rdkafka_assignment.o rdkafka_assignor.o rdkafka_range_assignor.o rdkafka_roundrobin_assignor.o rdkafka_sticky_assignor.o rdkafka_feature.o rdcrc32.o crc32c.o rdmurmur2.o rdfnv1a.o cJSON.o rdaddr.o rdrand.o rdlist.o tinycthread.o tinycthread_extra.o rdlog.o rdstring.o rdkafka_event.o rdkafka_metadata.o rdregex.o rdports.o rdkafka_metadata_cache.o rdavl.o rdkafka_sasl.o rdkafka_sasl_plain.o rdkafka_interceptor.o rdkafka_msgset_writer.o rdkafka_msgset_reader.o rdkafka_header.o rdkafka_admin.o rdkafka_aux.o rdkafka_background.o rdkafka_idempotence.o rdkafka_cert.o rdkafka_txnmgr.o rdkafka_coord.o rdbase64.o rdvarint.o rdbuf.o rdmap.o rdunittest.o rdkafka_mock.o rdkafka_mock_handlers.o rdkafka_mock_cgrp.o rdkafka_error.o rdkafka_fetcher.o rdkafka_telemetry.o rdkafka_telemetry_encode.o rdkafka_telemetry_decode.o nanopb/pb_encode.o nanopb/pb_decode.o nanopb/pb_common.o opentelemetry/metrics.pb.o opentelemetry/common.pb.o opentelemetry/resource.pb.o rdkafka_sasl_cyrus.o rdkafka_sasl_scram.o rdkafka_sasl_oauthbearer.o snappy.o rdgz.o rdkafka_zstd.o rdhdrhistogram.o rdkafka_ssl.o rdhttp.o rdkafka_sasl_oauthbearer_oidc.o rdkafka_lz4.o rdxxhash.o rddl.o rdkafka_plugin.o -o librdkafka.1.dylib -L/usr/local/Cellar/lz4/1.9.4/lib -llz4 -lm -lcurl -L/usr/local/opt/zstd/lib -lzstd -lsasl2 -L/usr/local/Cellar/openssl@3/3.3.1/lib -lssl -L/usr/local/Cellar/openssl@3/3.3.1/lib -lcrypto -lz -ldl -lpthread
55 error cp librdkafka.1.dylib librdkafka-dbg.1.dylib
55 error [33mCreating static library librdkafka.a^[(B[m
55 error ar rcs librdkafka.a rdkafka.o rdkafka_broker.o rdkafka_msg.o rdkafka_topic.o rdkafka_conf.o rdkafka_timer.o rdkafka_offset.o rdkafka_transport.o rdkafka_buf.o rdkafka_queue.o rdkafka_op.o rdkafka_request.o rdkafka_cgrp.o rdkafka_pattern.o rdkafka_partition.o rdkafka_subscription.o rdkafka_assignment.o rdkafka_assignor.o rdkafka_range_assignor.o rdkafka_roundrobin_assignor.o rdkafka_sticky_assignor.o rdkafka_feature.o rdcrc32.o crc32c.o rdmurmur2.o rdfnv1a.o cJSON.o rdaddr.o rdrand.o rdlist.o tinycthread.o tinycthread_extra.o rdlog.o rdstring.o rdkafka_event.o rdkafka_metadata.o rdregex.o rdports.o rdkafka_metadata_cache.o rdavl.o rdkafka_sasl.o rdkafka_sasl_plain.o rdkafka_interceptor.o rdkafka_msgset_writer.o rdkafka_msgset_reader.o rdkafka_header.o rdkafka_admin.o rdkafka_aux.o rdkafka_background.o rdkafka_idempotence.o rdkafka_cert.o rdkafka_txnmgr.o rdkafka_coord.o rdbase64.o rdvarint.o rdbuf.o rdmap.o rdunittest.o rdkafka_mock.o rdkafka_mock_handlers.o rdkafka_mock_cgrp.o rdkafka_error.o rdkafka_fetcher.o rdkafka_telemetry.o rdkafka_telemetry_encode.o rdkafka_telemetry_decode.o nanopb/pb_encode.o nanopb/pb_decode.o nanopb/pb_common.o opentelemetry/metrics.pb.o opentelemetry/common.pb.o opentelemetry/resource.pb.o rdkafka_sasl_cyrus.o rdkafka_sasl_scram.o rdkafka_sasl_oauthbearer.o snappy.o rdgz.o rdkafka_zstd.o rdhdrhistogram.o rdkafka_ssl.o rdhttp.o rdkafka_sasl_oauthbearer_oidc.o rdkafka_lz4.o rdxxhash.o rddl.o rdkafka_plugin.o
55 error cp librdkafka.a librdkafka-dbg.a
55 error [31mWARNING:[33m librdkafka-static.a: No static libraries available/enabled for inclusion in self-contained static library librdkafka-static.a: this library will be identical to librdkafka.a^[(B[m
55 error [31mWARNING:[33m librdkafka-static.a: The following libraries were not available as static libraries and need to be linked dynamically: -llz4 -lm -lcurl -lzstd -lsasl2 -lssl -lcrypto -lz -ldl -lpthread^[(B[m
55 error cp librdkafka.a librdkafka-static.a
55 error cp librdkafka-dbg.a librdkafka-static-dbg.a
55 error cp librdkafka-static.a librdkafka-static-dbg.a
55 error [33mCreating librdkafka.dylib symlink^[(B[m
55 error rm -f "librdkafka.dylib" && ln -s "librdkafka.1.dylib" "librdkafka.dylib"
55 error [33mGenerating pkg-config file rdkafka.pc^[(B[m
55 error [33mGenerating pkg-config file rdkafka-static.pc^[(B[m
55 error [33mChecking librdkafka integrity^[(B[m
55 error librdkafka.1.dylib             [32mOK^[(B[m
55 error librdkafka.a                   [32mOK^[(B[m
55 error Symbol visibility              [32mOK^[(B[m
55 error g++ -MD -MP -gstrict-dwarf -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -Wcast-align -Wno-non-virtual-dtor -c RdKafka.cpp -o RdKafka.o
55 error rm 11a9e3388a67e1ca5c31c1d8da49cb6d2714eb41.intermediate
56 error gyp info it worked if it ends with ok
56 error gyp info using node-gyp@10.2.0
56 error gyp info using node@23.1.0 | darwin | x64
56 error gyp info find Python using Python version 3.11.2 found at "/Users/norald.alejo/.pyenv/versions/3.11.2/bin/python3"
56 error (node:76952) ExperimentalWarning: CommonJS module /Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/supports-color/index.js using require().
56 error Support for loading ES Module in require() is an experimental feature and might change at any time
56 error (Use `node --trace-warnings ...` to show where the warning was created)
56 error gyp http GET https://nodejs.org/download/release/v23.1.0/node-v23.1.0-headers.tar.gz
56 error gyp http 200 https://nodejs.org/download/release/v23.1.0/node-v23.1.0-headers.tar.gz
56 error gyp http GET https://nodejs.org/download/release/v23.1.0/SHASUMS256.txt
56 error gyp http 200 https://nodejs.org/download/release/v23.1.0/SHASUMS256.txt
56 error gyp info spawn /Users/norald.alejo/.pyenv/versions/3.11.2/bin/python3
56 error gyp info spawn args [
56 error gyp info spawn args '/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
56 error gyp info spawn args 'binding.gyp',
56 error gyp info spawn args '-f',
56 error gyp info spawn args 'make',
56 error gyp info spawn args '-I',
56 error gyp info spawn args '/Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka/build/config.gypi',
56 error gyp info spawn args '-I',
56 error gyp info spawn args '/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
56 error gyp info spawn args '-I',
56 error gyp info spawn args '/Users/norald.alejo/Library/Caches/node-gyp/23.1.0/include/node/common.gypi',
56 error gyp info spawn args '-Dlibrary=shared_library',
56 error gyp info spawn args '-Dvisibility=default',
56 error gyp info spawn args '-Dnode_root_dir=/Users/norald.alejo/Library/Caches/node-gyp/23.1.0',
56 error gyp info spawn args '-Dnode_gyp_dir=/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/node-gyp',
56 error gyp info spawn args '-Dnode_lib_file=/Users/norald.alejo/Library/Caches/node-gyp/23.1.0/<(target_arch)/node.lib',
56 error gyp info spawn args '-Dmodule_root_dir=/Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka',
56 error gyp info spawn args '-Dnode_engine=v8',
56 error gyp info spawn args '--depth=.',
56 error gyp info spawn args '--no-parallel',
56 error gyp info spawn args '--generator-output',
56 error gyp info spawn args 'build',
56 error gyp info spawn args '-Goutput_dir=.'
56 error gyp info spawn args ]
56 error gyp info spawn make
56 error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
56 error rdkafka_broker.c:839:13: warning: variable 'idx' set but not used [-Wunused-but-set-variable]
56 error   839 |         int idx = -1;
56 error       |             ^
56 error 1 warning generated.
56 error rdkafka_request.c:3268:29: warning: format specifies type 'long' but the argument has type 'int64_t' (aka 'long long') [-Wformat]
56 error  3267 |                             "Received session lifetime %ld ms from broker",
56 error       |                                                        ~~~
56 error       |                                                        %lld
56 error  3268 |                             session_lifetime_ms);
56 error       |                             ^~~~~~~~~~~~~~~~~~~
56 error ./rdkafka_int.h:973:66: note: expanded from macro 'rd_kafka_dbg'
56 error   973 |                                       (RD_KAFKA_DBG_##ctx), fac, __VA_ARGS__); \
56 error       |                                                                  ^~~~~~~~~~~
56 error 1 warning generated.
56 error rdkafka_cgrp.c:1521:13: warning: variable 'not_revoking' set but not used [-Wunused-but-set-variable]
56 error  1521 |         int not_revoking   = 0;
56 error       |             ^
56 error 1 warning generated.
56 error rdbuf.c:912:16: warning: variable 'remains' set but not used [-Wunused-but-set-variable]
56 error   912 |         size_t remains = size;
56 error       |                ^
56 error 1 warning generated.
56 error rdmap.c:498:13: warning: variable 'fails' set but not used [-Wunused-but-set-variable]
56 error   498 |         int fails = 0;
56 error       |             ^
56 error 1 warning generated.
56 error snappy.c:1440:9: warning: variable 'written' set but not used [-Wunused-but-set-variable]
56 error  1440 |         size_t written = 0;
56 error       |                ^
56 error 1 warning generated.
56 error RdKafka.cpp:29:10: fatal error: 'string' file not found
56 error    29 | #include <string>
56 error       |          ^~~~~~~~
56 error 1 error generated.
56 error make[2]: *** [RdKafka.o] Error 1
56 error make[1]: *** [libs] Error 2
56 error make: *** [11a9e3388a67e1ca5c31c1d8da49cb6d2714eb41.intermediate] Error 2
56 error gyp ERR! build error 
56 error gyp ERR! stack Error: `make` failed with exit code: 2
56 error gyp ERR! stack at ChildProcess.<anonymous> (/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:216:23)
56 error gyp ERR! System Darwin 24.1.0
56 error gyp ERR! command "/Users/norald.alejo/.nvm/versions/node/v23.1.0/bin/node" "/Users/norald.alejo/.nvm/versions/node/v23.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
56 error gyp ERR! cwd /Users/norald.alejo/Programming/aioc/test/node_modules/node-rdkafka
56 error gyp ERR! node -v v23.1.0
56 error gyp ERR! node-gyp -v v10.2.0
56 error gyp ERR! not ok
57 silly unfinished npm timer reify 1730928142966
58 silly unfinished npm timer reify:build 1730928143909
59 silly unfinished npm timer build 1730928143909
60 silly unfinished npm timer build:deps 1730928143910
61 silly unfinished npm timer build:run:install 1730928143910
62 silly unfinished npm timer build:run:install:node_modules/node-rdkafka 1730928143910
63 verbose cwd /Users/norald.alejo/Programming/aioc/test
64 verbose os Darwin 24.1.0
65 verbose node v23.1.0
66 verbose npm  v10.9.0
67 verbose exit 1
68 verbose code 1
69 error A complete log of this run can be found in: /Users/norald.alejo/.npm/_logs/2024-11-06T21_22_22_655Z-debug-0.log
```

I am running into the same problem with Node v23

```
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error rdkafka_broker.c:839:13: warning: variable 'idx' set but not used [-Wunused-but-set-variable]
npm error         int idx = -1;
npm error             ^
npm error 1 warning generated.
npm error rdkafka_request.c:3268:29: warning: format specifies type 'long' but the argument has type 'int64_t' (aka 'long long') [-Wformat]
npm error                             session_lifetime_ms);
npm error                             ^~~~~~~~~~~~~~~~~~~
npm error ./rdkafka_int.h:973:66: note: expanded from macro 'rd_kafka_dbg'
npm error                                       (RD_KAFKA_DBG_##ctx), fac, __VA_ARGS__); \
npm error                                                                  ^~~~~~~~~~~
npm error 1 warning generated.
npm error rdkafka_cgrp.c:1521:13: warning: variable 'not_revoking' set but not used [-Wunused-but-set-variable]
npm error         int not_revoking   = 0;
npm error             ^
npm error 1 warning generated.
npm error rdbuf.c:912:16: warning: variable 'remains' set but not used [-Wunused-but-set-variable]
npm error         size_t remains = size;
npm error                ^
npm error 1 warning generated.
npm error rdmap.c:498:13: warning: variable 'fails' set but not used [-Wunused-but-set-variable]
npm error         int fails = 0;
npm error             ^
npm error 1 warning generated.
npm error snappy.c:1440:9: warning: variable 'written' set but not used [-Wunused-but-set-variable]
npm error         size_t written = 0;
npm error                ^
npm error 1 warning generated.
npm error rdkafka_example.c:583:21: warning: variable 'sendcnt' set but not used [-Wunused-but-set-variable]
npm error                 int sendcnt = 0;
npm error                     ^
npm error 1 warning generated.
npm error incremental_alter_configs.c:213:28: warning: variable 'op_type' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
npm error                 } else if (!strcmp(alter_op_type_s, "DELETE")) {
npm error                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error incremental_alter_configs.c:221:42: note: uninitialized use occurs here
npm error                     config, config_name, op_type, config_value);
npm error                                          ^~~~~~~
npm error incremental_alter_configs.c:213:24: note: remove the 'if' if its condition is always true
npm error                 } else if (!strcmp(alter_op_type_s, "DELETE")) {
npm error                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error incremental_alter_configs.c:184:17: note: variable 'op_type' is declared here
npm error                 rd_kafka_AlterConfigOpType_t op_type;
npm error                 ^
npm error 1 warning generated.
npm error In file included from ../src/binding.cc:11:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/binding.h:13:
npm error In file included from ../../nan/nan.h:62:
npm error In file included from /Users/kirrg001/Library/Caches/node-gyp/23.0.0/include/node/node.h:73:
npm error In file included from /Users/kirrg001/Library/Caches/node-gyp/23.0.0/include/node/v8.h:23:
npm error In file included from /Users/kirrg001/Library/Caches/node-gyp/23.0.0/include/node/cppgc/common.h:8:
npm error /Users/kirrg001/Library/Caches/node-gyp/23.0.0/include/node/v8config.h:13:2: error: "C++20 or later required."
npm error #error "C++20 or later required."
npm error  ^
npm error In file included from ../src/binding.cc:11:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/binding.h:16:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/common.h:20:
npm error ../deps/librdkafka/src/rdkafka.h:8264:1: warning: 'const' type qualifier on return type has no effect [-Wignored-qualifiers]
npm error const int rd_kafka_TopicPartitionInfo_partition(
npm error ^~~~~~
npm error In file included from ../src/binding.cc:11:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/binding.h:18:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/config.h:21:
npm error /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/callbacks.h:45:44: error: no template named 'CopyablePersistentTraits' in namespace 'v8'; did you mean 'Nan::CopyablePersistentTraits'?
npm error   std::vector<v8::Persistent<v8::Function, v8::CopyablePersistentTraits<v8::Function> > > callbacks;  // NOLINT
npm error                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error                                            Nan::CopyablePersistentTraits
npm error ../../nan/nan.h:208:29: note: 'Nan::CopyablePersistentTraits' declared here
npm error template<typename T> struct CopyablePersistentTraits {
npm error                             ^
npm error In file included from ../src/binding.cc:11:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/binding.h:21:
npm error /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/producer.h:80:8: warning: 'ActivateDispatchers' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
npm error   void ActivateDispatchers();
npm error        ^
npm error /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/connection.h:72:16: note: overridden virtual function is here
npm error   virtual void ActivateDispatchers() = 0;
npm error                ^
npm error In file included from ../src/binding.cc:11:
npm error In file included from /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/binding.h:21:
npm error /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/producer.h:81:8: warning: 'DeactivateDispatchers' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
npm error   void DeactivateDispatchers();
npm error        ^
npm error /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka/src/connection.h:73:16: note: overridden virtual function is here
npm error   virtual void DeactivateDispatchers() = 0;
npm error                ^
npm error 3 warnings and 2 errors generated.
npm error make: *** [Release/obj.target/node-librdkafka/src/binding.o] Error 1
npm error gyp ERR! build error
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.<anonymous> (/Users/kirrg001/.nvm/versions/node/v23.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:216:23)
npm error gyp ERR! System Darwin 23.4.0
npm error gyp ERR! command "/Users/kirrg001/.nvm/versions/node/v23.0.0/bin/node" "/Users/kirrg001/.nvm/versions/node/v23.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm error gyp ERR! cwd /Users/kirrg001/Downloads/haha28392/node_modules/node-rdkafka
npm error gyp ERR! node -v v23.0.0
npm error gyp ERR! node-gyp -v v10.2.0
npm error gyp ERR! not ok
```


- node-rdkafka latest
- Mac OSX 14.4.1 (23E224)
- x64

I'm using Node v18.20.5 (`node-rdkafka: 2.16.1`) and still getting the same error as the OP.
```
RdKafka.cpp:29:10: fatal error: 'string' file not found
   29 | #include <string>
      |          ^~~~~~~~
    ```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -10,7 +10,7 @@ jobs:
   build:
     strategy:
       matrix:
-        node: [16, 18, 20, 21, 22]
+        node: [16, 18, 20, 21, 22, 23]
         os: [ubuntu-22.04]
         include:
           # single mac test due to minute multipliers

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -17,7 +17,6 @@ jobs:
           # https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions
           - node: 22
             os: macos-14
-          # single windows test due to node.js 14 node-gyp / vs 2022 issues
           - node: 22
             os: windows-2022
     runs-on: ${{ matrix.os }}

--- a/binding.gyp
+++ b/binding.gyp
@@ -43,7 +43,7 @@
               }
             ],
             'cflags_cc' : [
-              '-std=c++17'
+              '-std=c++20'
             ],
             'msvs_settings': {
               'VCLinkerTool': {
@@ -122,7 +122,7 @@
                 'OS=="linux"',
                 {
                   'cflags_cc' : [
-                    '-std=c++17'
+                    '-std=c++20'
                   ],
                   'cflags_cc!': [
                     '-fno-rtti'
@@ -140,7 +140,7 @@
                     ],
                     'OTHER_CPLUSPLUSFLAGS': [
                       '-I/usr/local/opt/openssl/include',
-                      '-std=c++17'
+                      '-std=c++20'
                     ],
                   },
                 }

--- a/package.json
+++ b/package.json
@@ -35,12 +35,12 @@
     "jsdoc": "^4.0.2",
     "jshint": "^2.10.1",
     "mocha": "^10.2.0",
-    "node-gyp": "^10.1.0",
+    "node-gyp": "^11.1.0",
     "toolkit-jsdoc": "^1.0.0"
   },
   "dependencies": {
     "bindings": "^1.3.1",
-    "nan": "^2.19.0"
+    "nan": "^2.22.0"
   },
   "engines": {
     "node": ">=16"

--- a/src/callbacks.cc
+++ b/src/callbacks.cc
@@ -57,7 +57,7 @@ Dispatcher::~Dispatcher() {
   if (callbacks.size() < 1) return;
 
   for (size_t i=0; i < callbacks.size(); i++) {
-    callbacks[i].Reset();
+    delete callbacks[i];
   }
 
   uv_mutex_destroy(&async_lock);
@@ -98,24 +98,22 @@ void Dispatcher::Dispatch(const int _argc, Local<Value> _argv[]) {
   }
 
   for (size_t i=0; i < callbacks.size(); i++) {
-    v8::Local<v8::Function> f = Nan::New<v8::Function>(callbacks[i]);
-    Nan::Callback cb(f);
-    cb.Call(_argc, _argv);
+    callbacks[i]->Call(_argc, _argv);
   }
 }
 
 void Dispatcher::AddCallback(const v8::Local<v8::Function> &cb) {
-  Nan::Persistent<v8::Function,
-                  Nan::CopyablePersistentTraits<v8::Function> > value(cb);
+  Nan::Callback *value = new Nan::Callback(cb);
   // PersistentCopyableFunction value(func);
   callbacks.push_back(value);
 }
 
 void Dispatcher::RemoveCallback(const v8::Local<v8::Function> &cb) {
   for (size_t i=0; i < callbacks.size(); i++) {
-    if (callbacks[i] == cb) {
-      callbacks[i].Reset();
+    if (callbacks[i]->GetFunction() == cb) {
+      Nan::Callback *found_callback = callbacks[i];
       callbacks.erase(callbacks.begin() + i);
+      delete found_callback;
       break;
     }
   }

--- a/src/callbacks.h
+++ b/src/callbacks.h
@@ -18,10 +18,6 @@
 #include "rdkafkacpp.h"
 #include "src/common.h"
 
-typedef Nan::Persistent<v8::Function,
-  Nan::CopyablePersistentTraits<v8::Function> > PersistentCopyableFunction;
-typedef std::vector<PersistentCopyableFunction> CopyableFunctionList;
-
 namespace NodeKafka {
 
 class KafkaConsumer;
@@ -42,7 +38,7 @@ class Dispatcher {
   void Deactivate();
 
  protected:
-  std::vector<v8::Persistent<v8::Function, v8::CopyablePersistentTraits<v8::Function> > > callbacks;  // NOLINT
+  std::vector<Nan::Callback*> callbacks;  // NOLINT
 
   uv_mutex_t async_lock;
 

